logic.py

```diff
@@ -457,17 +308,18 @@ def to_cnf(s):
 
 def eliminate_implications(s):
     "Change implications into equivalent form with only &, |, and ~ as logical operators."
+    s = expr(s)
     if not s.args or is_symbol(s.op):
         return s  # Atoms are unchanged.
     args = list(map(eliminate_implications, s.args))
     a, b = args[0], args[-1]
-    if s.op == '>>' or s.op == '==>':
+    if s.op == '==>':
```

Comment:
I think this should be >> instead of ==>
I would prefer using ==> internally instead of >> as it is more intuitive